File: ISL.ReIdentification.WebApps.Configurations/ISL.ReIdentification.Configurations.Server.Tests.Acceptance/Models/DelegatedAccesses/DelegatedAccess.cs
﻿// ---------------------------------------------------------
// Copyright (c) North East London ICB. All rights reserved.
// ---------------------------------------------------------

using System;

namespace ISL.ReIdentification.Configurations.Server.Tests.Acceptance.Models.DelegatedAccesses
{
    public class DelegatedAccess
    {
        public Guid Id { get; set; }
        public string RequesterEmail { get; set; }
        public string RecipientEmail { get; set; }
        public bool IsDelegatedAccess { get; set; }
        public bool IsApproved { get; set; }
        public byte[]? Data { get; set; }
        public string IdentifierColumn { get; set; }
        public string CreatedBy { get; set; }
        public DateTimeOffset CreatedDate { get; set; }
        public string UpdatedBy { get; set; }
        public DateTimeOffset UpdatedDate { get; set; }
    }
}

File: ISL.ReIdentification.WebApps.Configurations/ISL.ReIdentification.Configurations.Client/src/models/delegatedAccessOLD/delegatedAccess.ts

export class ImpersonationContextOLD {
    public id: string;
    public requesterEmail: string;
    public recipientEmail: string;
    public isImpersonationContext: boolean;
    public isApproved: boolean;
    public data: Uint8Array;
    public identifierColumn: string;
    public createdBy?: string;
    public createdDate?: Date | undefined;
    public updatedBy?: string;
    public updatedDate?: Date | undefined;

    constructor(impersonationContext: ImpersonationContextOLD) {
        this.id = impersonationContext.id || crypto.randomUUID();
        this.requesterEmail = impersonationContext.requesterEmail || "";
        this.recipientEmail = impersonationContext.recipientEmail || "";
        this.isImpersonationContext = impersonationContext.isImpersonationContext || false;
        this.isApproved = impersonationContext.isApproved || false;
        this.data = impersonationContext.data || 0;
        this.identifierColumn = impersonationContext.identifierColumn || ""; 
        this.createdDate = impersonationContext.createdDate ? new Date(impersonationContext.createdDate) : undefined;
        this.createdBy = impersonationContext.createdBy || "";
        this.updatedDate = impersonationContext.updatedDate ? new Date(impersonationContext.updatedDate) : undefined;
        this.updatedBy = impersonationContext.updatedBy || "";
    }
}